In [1]:
import sqlite3 as sq

In [4]:
conn = sq.connect("chinook.db")
cursor = conn.cursor()

In [5]:
cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")

In [6]:
cursor.fetchall()

[('albums',),
 ('sqlite_sequence',),
 ('artists',),
 ('customers',),
 ('employees',),
 ('genres',),
 ('invoices',),
 ('invoice_items',),
 ('media_types',),
 ('playlists',),
 ('playlist_track',),
 ('tracks',),
 ('sqlite_stat1',)]

In [7]:
cursor.execute("SELECT * FROM employees;")

In [8]:
cursor.fetchall()

[(1,
  'Adams',
  'Andrew',
  'General Manager',
  None,
  '1962-02-18 00:00:00',
  '2002-08-14 00:00:00',
  '11120 Jasper Ave NW',
  'Edmonton',
  'AB',
  'Canada',
  'T5K 2N1',
  '+1 (780) 428-9482',
  '+1 (780) 428-3457',
  'andrew@chinookcorp.com'),
 (2,
  'Edwards',
  'Nancy',
  'Sales Manager',
  1,
  '1958-12-08 00:00:00',
  '2002-05-01 00:00:00',
  '825 8 Ave SW',
  'Calgary',
  'AB',
  'Canada',
  'T2P 2T3',
  '+1 (403) 262-3443',
  '+1 (403) 262-3322',
  'nancy@chinookcorp.com'),
 (3,
  'Peacock',
  'Jane',
  'Sales Support Agent',
  2,
  '1973-08-29 00:00:00',
  '2002-04-01 00:00:00',
  '1111 6 Ave SW',
  'Calgary',
  'AB',
  'Canada',
  'T2P 5M5',
  '+1 (403) 262-3443',
  '+1 (403) 262-6712',
  'jane@chinookcorp.com'),
 (4,
  'Park',
  'Margaret',
  'Sales Support Agent',
  2,
  '1947-09-19 00:00:00',
  '2003-05-03 00:00:00',
  '683 10 Street SW',
  'Calgary',
  'AB',
  'Canada',
  'T2P 5G3',
  '+1 (403) 263-4423',
  '+1 (403) 263-4289',
  'margaret@chinookcorp.com'),
 (5,


In [22]:

from langchain.sql_database import SQLDatabase
from langchain.agents import create_sql_agent
from langchain.memory import ConversationBufferMemory
from langchain_google_genai.llms import GoogleGenerativeAI



In [23]:
database = SQLDatabase.from_uri("sqlite:///C:/Users/hp/Desktop/python practice/Explore/day-06/chinook.db")

In [14]:
from dotenv import load_dotenv
import os

load_dotenv()

os.environ["GOOGLE_API_KEY"] = os.getenv("GEMINI_API_KEY")

llm = GoogleGenerativeAI(model="gemini-2.0-flash")

In [24]:
memory = ConversationBufferMemory()

In [25]:
agent = create_sql_agent(
    llm=llm,
    db=database,
    memory=memory
)

In [26]:
agent.run("fetch me details about Adam andrews")

'Andrew Adams is the General Manager. He was born on 1962-02-18 and hired on 2002-08-14. His address is 11120 Jasper Ave NW, Edmonton, AB, Canada, T5K 2N1. His phone number is +1 (780) 428-9482 and fax number is +1 (780) 428-3457. His email address is andrew@chinookcorp.com.'

IN DB:
(1,
  'Adams',
  'Andrew',
  'General Manager',
  None,
  '1962-02-18 00:00:00',
  '2002-08-14 00:00:00',
  '11120 Jasper Ave NW',
  'Edmonton',
  'AB',
  'Canada',
  'T5K 2N1',
  '+1 (780) 428-9482',
  '+1 (780) 428-3457',
  'andrew@chinookcorp.com'),

BY LLM:
Andrew Adams is the General Manager. He was born on 1962-02-18 and hired on 2002-08-14.
His address is 11120 Jasper Ave NW, Edmonton, AB, Canada, T5K 2N1. His phone number is +1 (780) 428-9482 and fax number is +1 (780) 428-3457.
His email address is andrew@chinookcorp.com.